raster/src/main/scala/geotrellis/raster/CellType.scala

```diff
@@ -316,15 +316,21 @@ object CellType {
       }
       IntUserDefinedNoDataCellType(ndVal.toInt)
     case ct if ct.startsWith("float32ud") =>
-      val ndVal = new Regex("\\d*.?\\d+$").findFirstIn(ct).getOrElse {
-        throw new IllegalArgumentException(s"Cell type $name is not supported")
+      try {
+        val ndVal = ct.stripPrefix("float32ud").toDouble.toFloat
```

Comment:
This is to keep in line with this approach:
https://github.com/geotrellis/geotrellis/blob/master/raster/src/main/scala/geotrellis/raster/io/geotiff/writer/TiffTagFieldValue.scala#L55